src/test/java/net/minecraftforge/debug/ConfigTest.java

```diff
@@ -99,4 +133,48 @@ public SubCat(String value)
             }
         }
     }
+
+    public static class ConfigurationGui extends GuiConfig
```

Comment:
That would not work because mainConfigGui would throw an AbstractMethodException.
Because the old implementors of this interface would not have that method existing at all.
What you could do is catch that exception and fall back silently to the old method.
That's the simplest way to do it. Or you could detect that the IModGuiFactory is missing that method {reflect the class at registration time to see if the method exists, or just try calling it and seeing if it AbtractErrors} and wrap it in another bounder class that implements mainConfigGui

Also, I would probably modify the name of that method to createConfigGui or something. A bit more specific on what the method should do.